Be clearer in definition of Inf and avoid relying on Modelica.Constants. (#3612)
modelica/ModelicaSpecification: chapters/classes.tex

  parameter StringType quantity    = "";
  parameter StringType unit        = "" "Unit used in equations";
  parameter StringType displayUnit = "" "Default display unit";

>   parameter RealType min = ..., max = ...; //  See below

Reviewer: Formatting:
```suggestion
  parameter RealType min = $\ldots$, max = $\ldots$; //  See below
```

Reviewer: But why have anything at all, considering that no default is given for `start`?
```suggestion
  parameter RealType min; // Minimum allowed value.
  parameter RealType max; // Maximum allowed value.
```

Reviewer: By the way, why do some of the attributes have explanations in comments, while some (`unit` and `displayUnit`) have description strings?

Author: > But why have anything at all, considering that no default is given for `start`?

I believe I tried that earlier, and @beutlich complained about that variant.
Since we have the assert using `min` I can see that it might require more.

As for `unit `and `displayUnit `having description strings and some attributes having comments: it has been that way for some time, I agree that it should likely be cleaned up at some point - it's just that we need reviews for existing topics before opening additional ones.

(Oh, and `start` not having any value was a very deliberate choice.)

>   parameter RealType min = $\ldots$, max = $\ldots$; //  See below

Reviewer: Perhaps the _See below_ goes without saying, and it would be of more interest to explain that these limit the value?
```suggestion
  parameter RealType min = $\ldots$; // Lower bound for $\langle$$\mbox{\emph{value}}$$\rangle$
  parameter RealType max = $\ldots$; // Upper bound for $\langle$$\mbox{\emph{value}}$$\rangle$
```

Also, if we now introduce the use of ellipsis for a default being explained in the text, shouldn't `fixed` be handled in the same style?
```
  parameter BooleanType fixed = $\ldots$; // Enforce exact value of 'start'
```
together with
> The default \lstinline!fixed!-attribute is \lstinline!true! for parameters and constants, and \lstinline!false! for other variables.

Author: Sure

\index{unbounded@\robustinline{unbounded}!attribute of \robustinline{Real}}%
\index{stateSelect@\robustinline{stateSelect}!attribute of \robustinline{Real}}

> The default \lstinline!min!- and \lstinline!max!-attributes do not constrain the value.

Reviewer 2: What are _default_ attributes? It can't be default values since there are no values assigned.

Author: I have tried to rewrite that, but still without introducing a special `+Inf` symbol.
If that fails we can just skip the last two commits.

> Therefore they do not constrain the value.

Reviewer: It would be worth pointing out that this still means that infinite values are outside the allowed range:
```suggestion
Therefore they only constrain the value to be finite, i.e., floating point representations of infinity are not allowed.
```

Author: I will reformulate - in my opinion the constraint to be an actual finite number is part of the type, and the default min/max do not restrict that further.

> \begin{nonnormative}

Reviewer: Just for better source code readability:
```suggestion

\begin{nonnormative}
```

Author: Did that as part of update

> The minimum representable floating point number is usually minus the maximum representable floating point number, and should not be confused with the minimum positive one.

Reviewer: I'm not sure _minus_ should be used like that; this looks better to me:
```suggestion
The minimum representable floating point number is usually the negation of the maximum representable floating point number, and should not be confused with the minimum positive one.
```

\index{unbounded@\robustinline{unbounded}!attribute of \robustinline{Real}}%
\index{stateSelect@\robustinline{stateSelect}!attribute of \robustinline{Real}}

> The default \lstinline!min! and \lstinline!max!-attributes are the minimum and maximum representable finite floating point numbers of the underlying type.

Reviewer: I think we tend do use suspended hyphenation.  Also, why not use `RealType`?
```suggestion
The default \lstinline!min!- and \lstinline!max!-attributes are the minimum and maximum representable finite floating point numbers of \lstinline!RealType!.
```

> The value must be a finite floating point number, and thus the default default \lstinline!min! and \lstinline!max!-attributes do not further constrain the value.

Reviewer: I guess you mean this:
```suggestion
The $\langle$$\mbox{\emph{value}}$$\rangle$ must be a finite floating point number, and thus the default \lstinline!min!- and \lstinline!max!-attributes do not impose any further constraints.
```

\begin{nonnormative}

> The minimum representable floating point number is usually the negation of the maximum representable floating point number, and should not be confused with the minimum positive one.

Reviewer: Perhaps better to avoid _one_ in a complicated sentence like this?
```suggestion
Of the representable floating point numbers, the minimum number is usually the negation of the maximum number, and should not be confused with the minimum positive number.
```

The following attributes shall be evaluable: \lstinline!quantity!, and \lstinline!fixed!.

> The default \lstinline!min! and \lstinline!max!-attributes are the minimum and maximum numbers of the underlying type.

Reviewer: As above:
```suggestion
The default \lstinline!min!- and \lstinline!max!-attributes are the minimum and maximum numbers of \lstinline!IntegerType!.
```
